pkg/skaffold/deploy/cloudrun/deploy.go

```diff
@@ -0,0 +1,247 @@
+	for _, manifest := range manifests {
+		if err := d.deployToCloudRun(ctx, out, manifest); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Dependencies list the files that would trigger a redeploy
+func (d *Deployer) Dependencies() ([]string, error) {
+	return []string{}, nil
+}
+
+// Cleanup deletes the created Cloud Run services
+func (d *Deployer) Cleanup(ctx context.Context, out io.Writer, dryRun bool, manifests manifest.ManifestList) error {
+	return d.deleteRunService(ctx, out, dryRun, manifests)
+}
+
+// Render writes out the k8s configs, we may want to support this with service configs in the future
+// but it's not being implemented now
+func (d *Deployer) Render(context.Context, io.Writer, []graph.Artifact, bool, string) error {
+	return nil
+}
+
+// GetDebugger Get the Debugger for Cloud Run. Not supported by this deployer.
+func (d *Deployer) GetDebugger() debug.Debugger {
+	return &debug.NoopDebugger{}
+}
+
+// GetLogger Get the logger for the Cloud Run deploy.
+func (d *Deployer) GetLogger() log.Logger {
+	return d.logger
+}
+
+// GetAccessor gets a no-op accessor for Cloud Run.
+func (d *Deployer) GetAccessor() access.Accessor {
+	return &access.NoopAccessor{}
+}
+
+// GetSyncer gets the file syncer for Cloud Run. Not supported by this deployer.
+func (d *Deployer) GetSyncer() sync.Syncer {
+	return &sync.NoopSyncer{}
+}
+
+// TrackBuildArtifacts is not supported by this deployer.
+func (d *Deployer) TrackBuildArtifacts([]graph.Artifact) {
+
+}
+
+// RegisterLocalImages is not supported by this deployer.
+func (d *Deployer) RegisterLocalImages([]graph.Artifact) {
+
+}
+
+// GetStatusMonitor gets the resource that will monitor deployment status.
+func (d *Deployer) GetStatusMonitor() status.Monitor {
+	return d.getMonitor()
+}
+
+func (d *Deployer) getMonitor() *Monitor {
+	if d.monitor == nil {
+		d.monitor = NewMonitor(d.labeller, d.clientOptions)
+	}
+	return d.monitor
+}
+func (d *Deployer) deployToCloudRun(ctx context.Context, out io.Writer, manifest []byte) error {
+	cOptions := d.clientOptions
+	if d.useGcpOptions {
+		cOptions = append(gcp.ClientOptions(ctx), cOptions...)
+	}
+	crclient, err := run.NewService(ctx, cOptions...)
+	if err != nil {
+		return sErrors.NewError(fmt.Errorf("unable to create Cloud Run Client"), &proto.ActionableErr{
+			Message: err.Error(),
+			ErrCode: proto.StatusCode_DEPLOY_GET_CLOUD_RUN_CLIENT_ERR,
+		})
+	}
+	service := &run.Service{}
+	if err = k8syaml.Unmarshal(manifest, service); err != nil {
+		return sErrors.NewError(fmt.Errorf("unable to unmarshal Cloud Run Service config"), &proto.ActionableErr{
+			Message: err.Error(),
+			ErrCode: proto.StatusCode_DEPLOY_READ_MANIFEST_ERR,
+		})
+	}
+	if service.Metadata.Namespace == "" {
+		service.Metadata.Namespace = d.DefaultProject
+	}
+
+	// we need to strip "skaffold.dev" from the run-id label because gcp labels don't support domains
+	runID, foundID := service.Metadata.Labels["skaffold.dev/run-id"]
```

Comment:
nit: skaffold also supports a -l flag that adds custom labels so custom labels might need to be checked here as well.

From - https://skaffold.dev/docs/references/cli/

```
  -l, --label=[]: Add custom labels to deployed objects. Set multiple times for multiple labels
```